client/state/login/magic-login/reducer.js

```diff
@@ -25,47 +33,14 @@ import {
 	SERIALIZE,
 } from 'state/action-types';
 
-export const isShowingRequestForm = createReducer( false, {
-	[ DESERIALIZE ]: () => false,
-	[ SERIALIZE ]: () => false,
-	[ MAGIC_LOGIN_HIDE_REQUEST_FORM ]: () => false,
-	[ MAGIC_LOGIN_SHOW_CHECK_YOUR_EMAIL_PAGE ]: () => false,
-	[ MAGIC_LOGIN_SHOW_REQUEST_FORM ]: () => true,
-	[ MAGIC_LOGIN_SHOW_INTERSTITIAL_PAGE ]: () => false,
-	[ MAGIC_LOGIN_SHOW_LINK_EXPIRED ]: () => false,
-} );
-
-export const isShowingInterstitialPage = createReducer( false, {
-	[ DESERIALIZE ]: () => false,
-	[ SERIALIZE ]: () => false,
-	[ MAGIC_LOGIN_HIDE_REQUEST_FORM ]: () => false,
-	[ MAGIC_LOGIN_HIDE_REQUEST_NOTICE ]: () => false,
-	[ MAGIC_LOGIN_SHOW_CHECK_YOUR_EMAIL_PAGE ]: () => false,
-	[ MAGIC_LOGIN_SHOW_INTERSTITIAL_PAGE ]: () => true,
-	[ MAGIC_LOGIN_SHOW_REQUEST_FORM ]: () => false,
-	[ MAGIC_LOGIN_SHOW_LINK_EXPIRED ]: () => false,
-} );
-
-export const isShowingExpiredPage = createReducer( false, {
-	[ DESERIALIZE ]: () => false,
-	[ SERIALIZE ]: () => false,
-	[ MAGIC_LOGIN_HIDE_REQUEST_FORM ]: () => false,
-	[ MAGIC_LOGIN_HIDE_REQUEST_NOTICE ]: () => false,
-	[ MAGIC_LOGIN_SHOW_CHECK_YOUR_EMAIL_PAGE ]: () => false,
-	[ MAGIC_LOGIN_SHOW_INTERSTITIAL_PAGE ]: () => false,
-	[ MAGIC_LOGIN_SHOW_REQUEST_FORM ]: () => false,
-	[ MAGIC_LOGIN_SHOW_LINK_EXPIRED ]: () => true,
-} );
-
-export const isShowingCheckYourEmailPage = createReducer( false, {
-	[ DESERIALIZE ]: () => false,
-	[ SERIALIZE ]: () => false,
-	[ MAGIC_LOGIN_HIDE_REQUEST_FORM ]: () => false,
-	[ MAGIC_LOGIN_HIDE_REQUEST_NOTICE ]: () => false,
-	[ MAGIC_LOGIN_SHOW_CHECK_YOUR_EMAIL_PAGE ]: () => true,
-	[ MAGIC_LOGIN_SHOW_INTERSTITIAL_PAGE ]: () => false,
-	[ MAGIC_LOGIN_SHOW_REQUEST_FORM ]: () => false,
-	[ MAGIC_LOGIN_SHOW_LINK_EXPIRED ]: () => false,
+export const showingView = createReducer( null, {
+	[ DESERIALIZE ]: () => null,
+	[ SERIALIZE ]: () => null,
```

Comment:
Per the implementation of `createReducer`, DE- and SERIALIZE don't need to be handled explicitly.